### PR TITLE
Repair OTP2 Middleware Support: send authorization header along with graphql request

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

The graphql request code incorrectly implemented `getOtpFetchOptions`, which caused the authorization header not to be sent. This PR corrects the flaw.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]

File: lib/actions/api.js [hunk 1]
```diff
@@ -279,7 +279,7 @@ export function createQueryAction(
     try {
       // Need to merge headers to support graphQL POST request with an api key
       const mergedHeaders = {
-        ...getOtpFetchOptions(state)?.headers,
+        ...getOtpFetchOptions(state, true)?.headers,
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/api.js:282]: This fixes a few other bugs as well where the middleware is acting funnily because no auth header is present

**Comment by miles-grant-ibigroup** [comment]
Since I'll be out tomorrow, assigning to both @amy-corson-ibigroup and @daniel-heppner-ibigroup and whoever approves it first can merge

**Comment by JymDyerIBI** [comment]
Ran this against sound-transit middleware and trips are being saved again.

**Review by amy-corson-ibigroup — APPROVED** [review]
Thanks!

**Pull request merged**